tests/http-adapter.spec.js

```diff
@@ -33,166 +29,252 @@ describe('HTTP adapter', function () {
+      fetchMock.mock(/.*api\/user\/2\//g, payload, {
+        method: 'PATCH'
+      })
 
-      return expect(adapter.update('/api/user/2/', payload)).resolves.toEqual(payload)
+      const response = await adapter.update('api/user/2/', payload)
+      expect(response).toEqual(payload)
+      done()
     })
   })
 
   describe('#destroy', function () {
-    it('deletes a record from the network', function () {
-      server.respondWith('DELETE', '/api/user/42/', [204, {}, ''])
-
-      return adapter.destroy('/api/user/42/')
+    it('deletes a record from the network', async done => {
+      fetchMock.reset()
+      fetchMock.mock(/.*api\/user\/42\//g, 204, {
+        method: 'DELETE'
+      })
+
+      const response = await adapter.destroy('api/user/42/')
+      expect(response).toMatchObject({})
+      done()
     })
   })
 
   describe('#serializeRequests', function () {
-    it('parallel requests do not start when `serializeRequests = true`', async function () {
-      server.autoRespond = false
-      server.respondImmediately = false
-
-      // Turn on `serializeRequests` and then make two parallel requests
-      adapter.serializeRequests = true
-      adapter.get('/api/user/42/')
-      adapter.get('/api/offer/9000/')
-
-      // Tick microtask queue
-      await Promise.resolve().then(() => {})
-
-      expect(server.requests).toHaveLength(1)
-      expect(server.requests[0].url).toEqual('/api/user/42/')
+    const path1 = 'api/user/42/'
+    const path2 = 'api/offer/9000/'
+    let fetchAPI
+    beforeEach(() => {
+      // set the fetch api to an interim state since we will be decorating it to run assertions
+      fetchAPI = global.fetch
     })
 
-    it('parallel requests do not start when `serializeRequests = true`, even if it was set after the pending request started', async function () {
-      server.autoRespond = false
-      server.respondImmediately = false
-
-      // Turn on `serializeRequests` after making the first request
-      adapter.get('/api/user/42/')
-      adapter.serializeRequests = true
-      adapter.get('/api/offer/9000/')
-
-      // Tick microtask queue
-      await Promise.resolve().then(() => {})
-
-      expect(server.requests).toHaveLength(1)
-      expect(server.requests[0].url).toEqual('/api/user/42/')
+    afterEach(() => {
+      // reset the mock state of fetch after each test
+      global.fetch = fetchAPI
     })
 
-    it('sequential requests start after previous finishes when `serializeRequests = true`', async function () {
-      server.autoRespond = false
-      server.respondImmediately = false
+    it('parallel requests do not start and start after previous finishes when `serializeRequests = true`', async done => {
+      const fetchMocked = fetchMock.sandbox().mock('*', {
+        status: 200,
+        method: 'GET',
+        body: {}
+      })
+      let numCalled = 0
+      
+      global.fetch = async function (url, init) {
```

Comment:
so this gets a bit funky when testing the serialize mode. Since I cant find too many tools that support faking requests for fetch, I went with a method that decorates the fetch API so we can run assertions on the data/requests. The methods are cleaned up after each test run. Can you all let me know what you think of this?